soroban-sdk-macros/src/lib.rs

```diff
@@ -95,6 +97,45 @@ pub fn contractimpl(_metadata: TokenStream, input: TokenStream) -> TokenStream {
     }
 }
 
+#[derive(Debug, FromMeta)]
+struct MetadataArgs {
+    key: String,
+    val: String,
+}
+
+#[proc_macro]
+pub fn metadata(metadata: TokenStream) -> TokenStream {
```

Comment:
Can we call this one `contractmetadata` or `contractmeta` for consistency with other macros?

Reply:
done